Util/src/main/java/io/deephaven/util/type/ArrayTypeUtils.java

```diff
@@ -426,70 +302,64 @@ public static Object[] getBoxedArray(Object value) {
             return getBoxedArray((byte[]) value);
-        } else if (c.equals(char[].class)) {
+        } else if (value instanceof char[]) {
             return getBoxedArray((char[]) value);
-        } else if (c.equals(double[].class)) {
+        } else if (value instanceof double[]) {
             return getBoxedArray((double[]) value);
-        } else if (c.equals(float[].class)) {
+        } else if (value instanceof float[]) {
             return getBoxedArray((float[]) value);
-        } else if (c.equals(int[].class)) {
+        } else if (value instanceof int[]) {
             return getBoxedArray((int[]) value);
-        } else if (c.equals(long[].class)) {
+        } else if (value instanceof long[]) {
             return getBoxedArray((long[]) value);
-        } else if (c.equals(short[].class)) {
+        } else if (value instanceof short[]) {
             return getBoxedArray((short[]) value);
         } else {
             return (Object[]) value;
         }
     }
 
     public static boolean equals(Object actualValue, Object expectedValue) {
-        final Class<?> ct = actualValue.getClass().getComponentType();
-        if (Object.class.isAssignableFrom(ct)) {
-            return Arrays.equals((Object[]) actualValue, (Object[]) expectedValue);
-        } else if (byte.class.isAssignableFrom(ct)) {
+        if (actualValue instanceof byte[] && expectedValue instanceof byte[]) {
             return Arrays.equals((byte[]) actualValue, (byte[]) expectedValue);
-        } else if (char.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof char[] && expectedValue instanceof char[]) {
             return Arrays.equals((char[]) actualValue, (char[]) expectedValue);
-        } else if (double.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof double[] && expectedValue instanceof double[]) {
             return Arrays.equals((double[]) actualValue, (double[]) expectedValue);
-        } else if (float.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof float[] && expectedValue instanceof float[]) {
             return Arrays.equals((float[]) actualValue, (float[]) expectedValue);
-        } else if (int.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof int[] && expectedValue instanceof int[]) {
             return Arrays.equals((int[]) actualValue, (int[]) expectedValue);
-        } else if (long.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof long[] && expectedValue instanceof long[]) {
             return Arrays.equals((long[]) actualValue, (long[]) expectedValue);
-        } else if (short.class.isAssignableFrom(ct)) {
+        } else if (actualValue instanceof short[] && expectedValue instanceof short[]) {
             return Arrays.equals((short[]) actualValue, (short[]) expectedValue);
+        } else {
+            return Arrays.equals((Object[]) actualValue, (Object[]) expectedValue);
```

Comment:
This last condition is breaking the pattern; the previous code was sort of checking it, otherwise returning false. In this code, could get a class cast exception. Looks like there is one caller, TableDiff.

Reply:
Technically could fail yes, but this method is expected to take an array. Could stick an if in there and return false (why would equals return false just because it isn't an array?), but TableDiff does verify that the other arg is an an array (and will fail to toString(...) the arg if it isn't an array... so this seems like a stretch).

Reply:
I think a `ClassCastException` is better than the previous `false`. If we really want to be more careful, we could check for `isArray()` and then throw `UnsupportedOperationException`.